public/app/features/dashboard/containers/EmbeddedDashboardPage.tsx

```diff
@@ -0,0 +1,148 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+
+import { GrafanaTheme2, PageLayoutType } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import { TimeZone } from '@grafana/schema';
+import { Button, ModalsController, PageToolbar, useStyles2 } from '@grafana/ui';
+import { Page } from 'app/core/components/Page/Page';
+import { useGrafana } from 'app/core/context/GrafanaContext';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import { useDispatch, useSelector } from 'app/types';
+
+import { updateTimeZoneForSession } from '../../profile/state/reducers';
+import { DashNavTimeControls } from '../components/DashNav/DashNavTimeControls';
+import { DashboardFailed } from '../components/DashboardLoading/DashboardFailed';
+import { DashboardLoading } from '../components/DashboardLoading/DashboardLoading';
+import { SaveDashboardDrawer } from '../components/EmbeddedDashboard/SaveDashboardDrawer';
+import { DashboardGrid } from '../dashgrid/DashboardGrid';
+import { DashboardModel } from '../state';
+import { initDashboard } from '../state/initDashboard';
+
+interface EmbeddedDashboardPageRouteParams {
+  uid: string;
+}
+
+interface EmbeddedDashboardPageRouteSearchParams {
+  callbackUrl?: string;
+  json?: string;
+  accessToken?: string;
+}
+
+export type Props = GrafanaRouteComponentProps<
+  EmbeddedDashboardPageRouteParams,
+  EmbeddedDashboardPageRouteSearchParams
+>;
+
+export default function EmbeddedDashboardPage({ route, queryParams }: Props) {
+  const dispatch = useDispatch();
+  const context = useGrafana();
+  const dashboardState = useSelector((store) => store.dashboard);
+  const dashboard = dashboardState.getModel();
+  const [dashboardJson, setDashboardJson] = useState('');
+
+  /**
+   * Create dashboard model and initialize the dashboard from JSON
+   */
+  useEffect(() => {
+    const callbackUrl = queryParams.callbackUrl;
+
+    if (!callbackUrl) {
+      throw new Error('No callback URL provided');
+    }
+    getBackendSrv()
+      .get(`${callbackUrl}/load-dashboard`)
```

Comment:
@Clarity-89 
I think this could be a pretty big security issue loading a resource from a URL provided via query params

Reply:
we're still working out all possible security issues, so I'll add this to the list.